lib/puma/minissl.rb

```diff
@@ -328,15 +331,15 @@ def to_io
       def accept
         @ctx.check
         io = @socket.accept
-        engine = Engine.server @ctx
+        engine = Engine.server (IS_JRUBY ? @ctx : @sslctx)
```

Comment:
this engine/sslctx is pretty hot-path, so I think we should instead just have an `engine` method which only has to run this check/ternary once. memoize or just use metaprogramming, your choice.

Reply:
Fixed, moved the JRuby checks to initialize...